lib/node.js

```diff
@@ -980,6 +980,9 @@ declare module "http" {
     options: Object | string,
     callback?: (response: IncomingMessage) => void
   ): ClientRequest;
+
+  declare var METHODS: Array<string>;
```

Comment:
I am out of capacity :-(

But it would be a simple copy-paste from:

* https://github.com/nodejs/node/blob/v7.8.0/lib/_http_server.js#L17-L80 and
* ... (looks like methods come from a native binding `http_parser`)